docs/api-docs/headless/end-to-end-guide/graphql.mdx

```diff
@@ -0,0 +1,756 @@
+
+<Callout type="info">
+#### Query best practices
+* Query less than 50 resources at a time.
+* Limit query complexity where possible.
+* Limit pagination where possible.
+* Employ security practices and limit scope where possible.
+* Employ caching where possible.
+</Callout>
+
+```graphql filename="Get a product" showLineNumbers copy
+# This query retrieves one product.
+query SingleProduct {
+  site {
+    product (entityId: ${params.id}) {
+      id
+      entityId
+      name
+      sku
+      description
+      prices {
+        price {
+          value
+          currencyCode
+        }
+      }
+    }
+  }
+}
+```
+
+## Creating a cart
+
+BigCommerce's GraphQL Storefront API provides the same cart and checkout objects as the REST Storefront API. The following example shows how to create a new cart by adding a simple product.
+
+<Tabs items={['Request', 'Response']}>
+  <Tab>
+    ```graphql filename="Example mutation: Create a cart (simple)" showLineNumbers copy 
+    # Creates a new cart, adding a simple product.
+    # In the GraphQL Playground, this will fail if you already have a cart.
+
+    mutation createCartSimple($createCartInput: CreateCartInput!) {
+      cart {
+        createCart(input: $createCartInput) {
+          cart {
+            entityId
+            lineItems {
+              physicalItems {
+                name
+                quantity
+              }
+              digitalItems {
+                name
+                quantity
+              }
+              giftCertificates {
+                name
+              }
+              customItems {
+                name
+                quantity
+              }
+            }
+          }
+        }
+      }
+    }
+    ```
+
+    ```json filename="GraphQL variables" showLineNumbers copy 
+    {
+      "createCartInput": {
+        "lineItems": [
+          {
+            "quantity": 1,
+            "productEntityId": 111
+          }
+        ]
+      }
+    }
+    ```
+  </Tab>
+  <Tab>
+    ```json filename="Example mutation: Create a cart (simple)" showLineNumbers copy
+    {
+      "data": {
+        "cart": {
+          "createCart": {
+            "cart": {
+              "entityId": "bb916deb-ddd6-4586-b65b-b8385e0e7a9d",
+              "lineItems": {
+                "physicalItems": [
+                  {
+                    "name": "[Sample] Smith Journal 13",
+                    "quantity": 1
+                  }
+                ],
+                "digitalItems": [],
+                "giftCertificates": [],
+                "customItems": []
+              }
+            }
+          }
+        }
+      }
+    }
+    ```
+  </Tab>
+</Tabs>
+
+## Add cart line items
```

Comment:
I think there's some confusion in this section.  There's just an example of a new set of GraphQL variables with the cart ID included, as if we're using the same mutation as the previous example.  It's not, though.  createCart and addLineItemsToCart are two different mutations; there should be a full example of the latter.

Reply:
Disregard the above.  I mis-read this section initially.